stdlib/effect.mli

```diff
@@ -24,6 +24,9 @@ external perform : 'a t -> 'a = "%perform"
 
     @raise Unhandled if there is no active handler. *)
 
+external fiber_id : unit -> int64 = "caml_fiber_id"
```

Comment:
Do we need to expose the int64 to the external interface? 

```
module Id : sig
  type t
  val compare : t -> t -> int
  val pp : ...
end
val fiber_id : unit -> Id.t
```

would be sufficient to be able to use them now and modify the internal representation (e.g. to an unboxed "Int63" on 64-bit platforms) in the future.

Reply:
It's safe to say that we don't know what a good API for unique identifiers looks like.  

In an attempt to defuse the issue, I'd like to point out that we do not need to expose fiber IDs to OCaml code in 5.0.  Right now the only potential use is for the bytecode debugger, which will access these IDs through a completely different, RPC-based API.  So, why don't we focus on the C implementation of fiber IDs and on the debugger interface first?

Reply:
> In an attempt to defuse the issue, I'd like to point out that we do not need to expose fiber IDs to OCaml code in 5.0. Right now the only potential use is for the bytecode debugger, which will access these IDs through a completely different, RPC-based API. So, why don't we focus on the C implementation of fiber IDs and on the debugger interface first?

I agree that allowing the debugger PR to complete is the most important thing.

I don't even think it's necessary to expose these fiber ids to `ocamldebug`: `trap barriers` are only used to `finish` the current function, which should be the outermost one in the last active fiber. Only the runtime needs to be able to identify the exact fiber, so it shouldn't be necessary to communicate the id to `ocamldebug`. Or did I miss something import with `checkpoint` and `trap barrier`?